libraries/rush-lib/src/cli/actions/BaseInstallAction.ts

```diff
@@ -244,9 +244,12 @@ export abstract class BaseInstallAction extends BaseRushAction {
         for (const selectedSubspace of selectedSubspaces) {
           installManagerOptions.subspace = selectedSubspace;
           if (selectedSubspace.getPnpmOptions()?.alwaysFullInstall) {
-            installManagerOptions.filteredProjects = [];
+            installManagerOptions.filteredProjects = undefined;
           } else {
-            installManagerOptions.filteredProjects = filteredProjectsForSubspace.get(selectedSubspace) || [];
+            const filteredProjects: Set<RushConfigurationProject> | undefined =
+              filteredProjectsForSubspace.get(selectedSubspace);
```

Comment:
Rename `filteredProjectsForSubspace` -> `filteredProjectsBySubspace` to follow or naming convention.  Without that, I had trouble realizing that `filteredProjects` cannot include projects from other subspaces.